tests/test_socket_server.py

```diff
@@ -93,14 +97,14 @@ async def test_disconnect(self) -> None:
         assert self.srv.connected
         await self.write(command=common.Command.DISCONNECT, parameters={})
         # Give time to the socket server to clean up internal state and exit.
-        await asyncio.sleep(0.5)
+        await asyncio.sleep(SOCKET_SERVER_RESPOND_TIME)
```

Comment:
I gently suggest awaiting a task with a suitable time limit. It's more robust in that you can use a long enough time limit that even if Jenkins is bogged down the test still works, while also allowing you to exit as soon as the socket is closed. However, it does require more code and it may be more fuss than it's worth.

One solution is to use a connect callback and have it set a future variable's result (if the future is not already done). Before you write the disconnect command, set that future variable to a new future. Then await it (with wait_for to provide the time limit). This is a arguably a reasonable use case for a test case method for the callback and a test case instance variable for the future, though a locally defined function would also work. What I have in mind is:
```
def asyncSetup(self):
    self.connected_future = asyncio.Future()
    self.serv = ...(..., connect_callback=self.connect_callback, ...)
    ...

def connect_callback(self, server):
    if not self.connected_future.done():
        self.connected_future.set_result(server.connected)

def test_disconnect(self):
    ...
    self.connected_future = asyncio.Future()
    await asyncio.sleep(SOCKET_SERVER_RESPOND_TIME)
    assert not await asyncio.wait_for(self.connected_future, SOCKET_SERVER_RESPOND_TIME)
```

Reply:
Thanks for this great suggestion. I had to make sure that the `connect_callback` in the test class calls the `connect_callback` in the SocketServer class, since that starts the `read_loop_task` when a client connects. Once done it works beautifully well. This is a great design pattern that I keep forgetting about. I made note of it now so I can use it in other places as well.